src/NServiceBus.AzureStoragePersistence.SagaDeduplicator/Extensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus.AzureStoragePersistence.SagaDeduplicator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.WindowsAzure.Storage.Table;
+
+    public static class Extensions
+    {
+        public static EdmType GetPropertyType(this CloudTable table, string propertyName)
+        {
+            var query = new TableQuery
+            {
+                TakeCount = 1
+            };
+            EntityProperty propertyValue;
+            var entity = table.ExecuteQuery(query).SingleOrDefault();
```

Comment:
Are we certain that all entities in the table have the same schema? from a tablestorage perspective this is not required and if a field is null it will not appear as a property on the record. So just grabbing the first one might be a mistake, if the property value was null, then the property would not exist on that record.